ssl/ssl_lib.c

```diff
@@ -3699,7 +3699,7 @@ SSL_CTX *SSL_CTX_new_ex(OSSL_LIB_CTX *libctx, const char *propq,
     ret->session_cache_mode = SSL_SESS_CACHE_SERVER;
     ret->session_cache_size = SSL_SESSION_CACHE_MAX_SIZE_DEFAULT;
     /* We take the system default. */
-    ret->session_timeout = meth->get_timeout();
+    ret->session_timeout = ossl_seconds2time(meth->get_timeout());
```

Comment:
I would actually suggest changing `get_timeout()` functions to return an `OSSL_TIME` (See #19107)